### PR TITLE
feature(Source): add encoding parameter for x-bil data

### PR DESCRIPTION
**Pull request by mgermerie**
## Description

Add `encoding` property to `Source`. This property defines the number of bits the source data are encoded on. It only works with `image/x-bil` source data.

### CODE REVIEW

**Review by gchoqueux — COMMENTED** [review]
* handling deprecated  `useColorTextureElevation`
* add documentation

File: examples/mars.html [hunk 1]
```diff
@@ -74,6 +73,7 @@
                     crs: 'EPSG:4326',
                     url : 'https://api.nasa.gov/mars-wmts/catalog/Mars_MGS_MOLA_DEM_mosaic_global_463m_8/1.0.0//default/default028mm/${z}/${y}/${x}.png',
                     zoom: { min: 0, max: 5 },
+                    encoding: 'RGB',
```
Comment (gchoqueux) [inline examples/mars.html:76]: It's not `RGB` but grey rgb

File: src/Core/View.js [hunk 2]
```diff
@@ -311,7 +311,8 @@ class View extends THREE.EventDispatcher {
                 } else {
                     return layer._reject(new Error(`Cant add color layer ${layer.id}: the maximum layer is reached`));
                 }
-            } else if (layer.isElevationLayer && layer.source.format == 'image/x-bil;bits=32') {
+            } else if (layer.isElevationLayer) {
+                layer.source.networkOptions.encoding = layer.source.encoding;
```
Comment (gchoqueux) [inline src/Core/View.js]: I think it will better to add a new `fileOptions`.

File: src/Layer/ElevationLayer.js [hunk 3]
```diff
@@ -53,12 +51,20 @@ class ElevationLayer extends RasterLayer {
      * view.addLayer(elevation);
      */
     constructor(id, config = {}) {
+        if (config.useColorTextureElevation !== undefined) {
```
Comment (gchoqueux) [inline src/Layer/ElevationLayer.js:54]: use undeprecator

**Comment by Desplandis** [comment]
@mgermerie Have we encounter some use-cases where elevation data is provided as 8bits or 16bits BIL? I think we'll depreciate the BIL format in future versions in favor of more "standard" formats (replacing it with COG and Mapbox RGB). So this PR should be closed? WDYT?

**Comment by mgermerie** [comment]
I must admit I don't remember the use-case. Maybe @gchoqueux remembers ?

Otherwise, should we drop BIL support, I agree to close the PR.

**Pull request closed without merging**